tests/util/minikube/cmd/minikube.go

```diff
@@ -0,0 +1,174 @@
+		"Insecure Docker registries to pass to the Docker daemon.  The default service CIDR range will automatically"+
+			" be added.")
+	setupCmd.PersistentFlags().StringVar(&sa.vmDriver, "vm-driver", "none", "vm driver for minikube"+
+		" in Istio. Supported are none, hyperkit and kvm2")
+	setupCmd.PersistentFlags().Uint16Var(&sa.remoteRegistryPort, "remoteRegistryPort", 5000, "port"+
+		" number where registry can be port-forwarded on user's machine. Default is 5000, same as local registry port.")
+	setupCmd.PersistentFlags().StringVar(&sa.kubeconfig, "kubeconfig", "",
+		"Use a Kubernetes configuration file instead of in-cluster configuration")
+	return setupCmd
+}
+
+func teardownCfgCmd(rawArgs []string, printf, fatalf shared.FormatFn) *cobra.Command {
+	sa := &minikubeSetupArgs{}
+	teardownCmd := &cobra.Command{
+		Use:   "delete",
+		Short: "deletes minikube environmnet",
+		Run: func(cmd *cobra.Command, args []string) {
+			deleteMinikubeEnvironment(sa, printf, fatalf)
+		},
+	}
+	teardownCmd.PersistentFlags().StringVar(&sa.kubeconfig, "kubeconfig", "",
+		"Use a Kubernetes configuration file instead of in-cluster configuration")
+	return teardownCmd
+}
+
+// setupMinikubeEnvironment starts minikube and sets up local registry in minikube's k8's cluster.
+func setupMinikubeEnvironment(sa *minikubeSetupArgs, printf, fatalf shared.FormatFn) {
+	goPath := os.Getenv("GOPATH")
+	if len(goPath) == 0 {
+		fatalf("GOPATH not set.")
+	}
+
+	startMinikube(sa, printf, fatalf)
+	setupDockerToTalkToMinikube(printf, fatalf)
+	checkMinikubeRunning(fatalf)
+
+	printf("#Setting up in-cluster local registry in minikube kuberntes cluster#")
+	cmd.SetupLocalRegistry(sa.remoteRegistryPort, sa.kubeconfig, printf, fatalf)
+	printf("")
+	printf("#Minikube Setup is done#")
+	printf("#Please export HUB=localhost:%d in your console#", sa.remoteRegistryPort)
+}
+
+func startMinikube(sa *minikubeSetupArgs, printf, fatalf shared.FormatFn) {
+	printf("#Starting Minikube#")
+	startMinikubeCmd := fmt.Sprintf("minikube start --kubernetes-version=%s --cpus=%d --memory=%d "+
+		"--vm-driver=%s --extra-config=%s --insecure-registry=%s", sa.kubernetesVersion, sa.cpus, sa.memory, sa.vmDriver,
+		strings.Join(sa.extraOptions, " --extra-config="), strings.Join(sa.insecureRegistry, " --insecure-registry="))
+	if _, err := util.Shell(startMinikubeCmd); err != nil {
+		fatalf("Failed starting minikube: %v", err)
+	}
+}
+
+func deleteMinikube(sa *minikubeSetupArgs, printf, fatalf shared.FormatFn) {
+	if _, err := util.Shell("minikube delete"); err != nil {
+		fatalf("Failed deleting minikube: %v", err)
+	}
+}
+
+func setupDockerToTalkToMinikube(printf, fatalf shared.FormatFn) {
+	printf("Running Command: eval $(minikube docker-env)")
+	c := exec.Command("bash", "-c", "eval $(minikube docker-env)")
+	if err := c.Run(); err != nil {
+		fatalf("Failed setting up docker environment to talk to minikube: %v", err)
+	}
+}
+
+func cleanupDockerToTalkToMinikube(printf, fatalf shared.FormatFn) {
```

Comment:
How does it even work ? The eval will happen in a child, having no effect on the parent AFAIK.

Reply:
yes, I have to remove this function.. this was a cl just to get initial feedback.. 

I will author a separate cl which will use docker client to achieve what is happening in eval here
(https://godoc.org/github.com/docker/docker/client#FromEnv)